cocos/core/value-types/enum.ts

```diff
@@ -29,6 +29,24 @@ import { legacyCC } from '../global-exports';
 import { errorID } from '../platform/debug';
 import { assertIsTrue } from '../data/utils/asserts';
 
+type EnumType = Record<string, string | number>;
+
+interface EnumRegister {
+    /**
+     * `string` indicates the enumeration's name. `undefined` indicates the enumeration has no name.
+     */
+    name: string | undefined;
+
+    /**
+     * `null` if haven't not computed.
+     */
+    enumerators: null | Enum.Enumerator<EnumType>[];
+}
+
+const enumTypeRegistry = new WeakMap<EnumType, EnumRegister | null>();
+
+const enumNameReverseLookupMap = new Map<string, EnumType>();
```

Comment:
I don't think there's a need to emphasize "reverse" here. Since it's a register, it's obviously mapping from name to type.

Reply:
Fixed in d209f889ed711c60c2ab5c84ba899347aaadb5d8